### PR TITLE
Improve code reusability and implement 2 controller test cases

### PR DESCRIPTION
**Pull request by GitH3ll**

### CODE REVIEW

File: pkg/controller/controller_test.go [hunk 1]
```diff
@@ -1 +1,240 @@
...
+			Winner:  "",
+			Users: map[string]struct {
+			}{
+				okUser1: {},
+				okUser2: {},
+			},
+			Prize: 400}, nil)
+	dbMock.On("RemoveTournament", mock.Anything, okTournamentID).
+		Return(nil)
+	dbMock.On("AddUserBalance", mock.Anything, mock.AnythingOfType("string"), 400).
```
Comment (GitH3ll) [inline pkg/controller/controller_test.go]: How should I should I check winner's ID? Winner is chosen randomly inside FinishTournament func.

Comment (sashayakovtseva) [reply pkg/controller/controller_test.go]: Check non-empty

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -126,20 +127,23 @@ func (c *Control) FinishTournament(ctx context.Context, id string) error {
 	if err != nil {
 		return errors.Wrap(err, "couldn't get tournament data")
 	}
+	users := mapToSlice(tournament.Users)
+	if len(users) <= 0 {
```
Comment (sashayakovtseva) [inline pkg/controller/controller.go]: Len can be checked before transform to slice

File: pkg/controller/controller_test.go [hunk 3]
```diff
@@ -1 +1,240 @@
...
+	}{
+		{
+			caseName:         "ok",
+			userID:           randomID,
+			pointsToSubtract: 400,
+		},
+		{
+			caseName:         "user error",
+			userID:           randomID[3:],
+			pointsToSubtract: math.MaxInt32,
```
Comment (sashayakovtseva) [inline pkg/controller/controller_test.go]: Why?)

**Pull request merged**